app/app.go

```diff
@@ -361,18 +361,32 @@ func NewOsmosisApp(
 	// there is nothing left over in the validator fee pool, so as to keep the
 	// CanWithdrawInvariant invariant.
 	// NOTE: staking module is required if HistoricalEntries param > 0
+	// NOTE: capability module's beginblocker must come before any modules using capabilities (e.g. IBC)
 	app.mm.SetOrderBeginBlockers(
 		// Upgrades should be run _very_ first
 		upgradetypes.ModuleName,
 		// Note: epochs' begin should be "real" start of epochs, we keep epochs beginblock at the beginning
 		epochstypes.ModuleName,
-		minttypes.ModuleName, poolincentivestypes.ModuleName, distrtypes.ModuleName, slashingtypes.ModuleName,
-		evidencetypes.ModuleName, stakingtypes.ModuleName, ibchost.ModuleName, capabilitytypes.ModuleName,
+		capabilitytypes.ModuleName, minttypes.ModuleName,
+		poolincentivestypes.ModuleName,
+		distrtypes.ModuleName, slashingtypes.ModuleName,
+		evidencetypes.ModuleName, stakingtypes.ModuleName, ibchost.ModuleName, ibctransfertypes.ModuleName,
+		authtypes.ModuleName, banktypes.ModuleName, govtypes.ModuleName, crisistypes.ModuleName, genutiltypes.ModuleName,
+		authz.ModuleName,
+		paramstypes.ModuleName, vestingtypes.ModuleName,
+		gammtypes.ModuleName, incentivestypes.ModuleName, lockuptypes.ModuleName, claimtypes.ModuleName,
+		poolincentivestypes.ModuleName, superfluidtypes.ModuleName, bech32ibctypes.ModuleName, txfeestypes.ModuleName,
```

Comment:
Was it a bug before that we didn't have everything listed, or did not listing them here mean their begin block could occur in any order after everything ordered was done?

Ditto question for end blocker

Reply:
This was a new requirement with 0.45 to avoid things being unlisted.

Before you only listed the ones being run. Now you must list all or it will panic on startup

Reply:
oh cool, thanks for the explanation! Didn't know that

Reply:
Yeah, idk how I feel about this.  Seems to just make it more messy imo...

Reply:
Yeah, this whole explicit singly ordered list seems like an unscalable and error prone method.

I hope with the new app wiring work, theres a way for modules to declare what they need to come after, and then app.go linearizes the init genesis' itself. I should ask whats the plan there